e2e/playwright/point-click.spec.ts

```diff
@@ -2812,4 +2812,118 @@ radius = 8.69
       expect(editor.expectEditor.toContain(newCodeToFind)).toBeTruthy()
     })
   })
+
+  test(`Set appearance`, async ({
+    context,
+    page,
+    homePage,
+    scene,
+    editor,
+    toolbar,
+    cmdBar,
+  }) => {
+    const initialCode = `sketch001 = startSketchOn('XZ')
+profile001 = circle({
+  center = [0, 0],
+  radius = 100
+}, sketch001)
+extrude001 = extrude(profile001, length = 100)
+`
+    await context.addInitScript((initialCode) => {
+      localStorage.setItem('persistCode', initialCode)
+    }, initialCode)
+    await page.setBodyDimensions({ width: 1000, height: 500 })
+    await homePage.goToModelingScene()
+    await scene.waitForExecutionDone()
+
+    // One dumb hardcoded screen pixel value
+    const testPoint = { x: 500, y: 250 }
+    const initialColor: [number, number, number] = [135, 135, 135]
+
+    await test.step(`Confirm extrude exists with default appearance`, async () => {
+      await toolbar.closePane('code')
+      await scene.expectPixelColor(initialColor, testPoint, 15)
+    })
+
+    async function goThroughFlowAndCheckExecution(
+      option: string,
+      hex: string,
+      shapeColor: [number, number, number]
+    ) {
+      await toolbar.openPane('feature-tree')
+      const operationButton = await toolbar.getFeatureTreeOperation(
+        'Extrude',
+        0
+      )
+      await operationButton.click({ button: 'right' })
+      await page.waitForTimeout(500) // just cause it's nice to see
+      const menuButton = page.getByTestId('context-menu-set-appearance')
+      await menuButton.click()
+      await cmdBar.expectState({
+        commandName: 'Appearance',
+        currentArgKey: 'color',
+        currentArgValue: '',
+        headerArguments: {
+          Color: '',
+        },
+        highlightedHeaderArg: 'color',
+        stage: 'arguments',
+      })
+      const item = page.getByText(option, { exact: true })
+      await page.waitForTimeout(500) // just cause it's nice to see
+      await item.click()
+      await cmdBar.expectState({
+        commandName: 'Appearance',
+        headerArguments: {
+          Color: hex,
+        },
+        stage: 'review',
+      })
+      await cmdBar.progressCmdBar()
+      await toolbar.closePane('feature-tree')
+      await scene.expectPixelColor(shapeColor, testPoint, 20)
+      await toolbar.openPane('code')
+      if (hex === 'default') {
+        const anyAppearanceDeclaration = `|> appearance(`
+        await editor.expectEditor.not.toContain(anyAppearanceDeclaration)
+      } else {
+        const declaration = `|> appearance(%, color = '${hex}')`
+        await editor.expectEditor.toContain(declaration)
+        // TODO: fix selection range after appearance update
+        // await editor.expectState({
+        //   diagnostics: [],
+        //   activeLines: [declaration],
+        //   highlightedCode: '',
+        // })
+      }
+      await page.waitForTimeout(1000) // just cause it's nice to see
```

Comment:
This timeout might add up, it's really just like 10 additional seconds though I guess

Reply:
Whoops that was for my testing let me 🔪 them all